GitCommands/Git/Commands/GitCommandHelpers.cs

```diff
@@ -48,16 +59,36 @@ public static ArgumentString SubmoduleSyncCmd(string? name)
             };
         }
 
-        public static ArgumentString AddSubmoduleCmd(string remotePath, string localPath, string branch, bool force)
+        public static ArgumentString AddSubmoduleCmd(string remotePath, string localPath, string branch, bool force, IEnumerable<GitConfigItem> configs = null)
         {
-            return new GitArgumentBuilder("submodule")
+            GitArgumentBuilder argsBuilder = new("submodule")
             {
                 "add",
                 { force, "-f" },
                 { !string.IsNullOrEmpty(branch), $"-b \"{branch?.Trim()}\"" },
                 remotePath.ToPosixPath().Quote(),
                 localPath.ToPosixPath().Quote()
             };
+
+            if (configs != null)
+            {
+                foreach (var cfg in configs)
+                {
+                    argsBuilder.Add(cfg);
+                }
+            }
+
+            return argsBuilder;
+        }
+
+        /// <summary>
+        /// Gets <see cref="GitConfigItem"/> that sets 'protocol.file.allow' to always.
+        /// </summary>
+        /// <remarks>IEnumerable to allow future concat with other needed configs.</remarks>
+        /// <returns>Config with 'protocol.file.allow set to always</returns>
+        public static IEnumerable<GitConfigItem> GetAllowFileConfig()
+        {
+            yield return new GitConfigItem(SettingKeyString.AllowFileProtocol, "always");
```

Comment:
Please read all the linked reference material in issue and all my comments in pull request. I did a pretty good job of documenting the issue and what the config does and even using in a config file doesn't solve the problem.

Reply:
My understanding is if config is added in version x, then <x ignores config.

Reply:
What if you specify it explicitly on the command line?

Reply:
I have done a ton of testing. If you doubt the work pull it down and rub your own tests to verify. I have already hit it hard and validated the fix.

Reply:
This will lay the foundation for a used allows blaa functionality down the road.  This just deals with getting tests green. Run the noted tests before this work on 2.38.1 git version. Take off the error swallowing and see the problem yourself.

Reply:
And I mean actually debug a test and not just run a test.

Reply:
https://github.com/gitextensions/gitextensions/pull/10389#issuecomment-1312813174

Reply:
Any git config that is used that is not looked for is ignored.  You could setup 

```
git config mstv.seceret ssshhhh
````
and git would ignore because it does nothing for git.  Same principle for a config value that is looked for in later versions and not older.  So before this config was added, git is going to say so what.  I don't care.  When it looks for it, command line is just another level of config that is lower in hierarchy of all the config files like --system, --global, --worktree, and --local.  So command line is the lowest and overrides all config file values when passed in.

Reply:
https://bugs.launchpad.net/ubuntu/+source/git/+bug/1993586

Reply:
My concern was: "What happens if an older git version is installed and such options are added to the above submodule commands?"
I had not noticed that `GetAllowFileConfig()` is used by tests only. It would have been clearer if this function was not be located in application code.

Reply:
For the test execution we can demand a recent git version, i.e. nothing to do - perhaps except moving the function to a test helper.

Reply:
It will be used down the road in program too.  This is setting up plumbing.  The gui will pass down stuff and that was a plumbing feature.  The how is the config being passed in.  The when it is passed in and whatnot will be  determined by gui flow logic in next pull request.